mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown

```diff
@@ -112,6 +112,12 @@ There are now three label-related fields with the new model:
 
 After upgrading to `5.0.0`, and then running `terraform refresh` or `terraform apply`, these three fields should show in the state file of the resources with a self-applying `labels` field.
 
+## Provider Default Zone and Region Values
+
+Several tests were inconsistent with one another when it came to testing `location`, `zone`, and `region` values in the provider block. The inconsistencies included some tests not working with self-links while others did. This has been addressed while also addressing the use of the `region` value within the provider block.
+
+Having the `region` value set in both the resource and provider block will now result in the use of the `region` value being used from the provider block. Prior to the 5.0.0, the `region` value would be ignored and the resource `region` value would be used instead. The same can be said with `region` and `zone` values in the provider where the `region` value, despite being set, would be ignored with the `zone` value being used instead. Users should also expect to see `region` being used if both `region` and `zone` values are set in a resource.
```

Comment:
Of the changes we've introduced I think that the only really important one is that a resource with a `location` field and no location info set in the resource block would previously use the zone in the provider block. After this change that resource's location info would come from the provider's region value (if set) and the default region could be a different region than the region that contains the default zone.

For other changes -particularly the 'you couldn't use a self link here but now can' ones- I'm not sure how to describe those changes to users. We could ask for more input from the Google team on this.

What do you think?

Reply:
It seems like its not necessary to include `self-links` in the guide since regardless if they are using self-links or not if they do migrate to 5.0.0 they shouldn't expect in sudden changes from it. Although it would be good to get some input from the Google team on this.